unit-tests/test-fw-update.py

```diff
@@ -75,23 +75,30 @@ def reset_update_counter( device ):
     send_hardware_monitor_command( device, cmd )
 
 def find_image_or_exit( product_name, image_mask ):
-    pattern = re.compile('Intel RealSense ')
+    pattern = re.compile(r'^Intel RealSense ((\S+?)(\d+)(\S*))')
     match = pattern.search(product_name)
     if not match:
-        # Didn't find 'Intel RealSense ' in product_name
-        log.f(product_name, " is not valid")
+        log.f(product_name, " is not valid - Didn't find 'Intel RealSense ' in product_name")
 
-    # finding file containing image for FW update
+    # Finding file containing image for FW update
+    # For example: for 'Intel RealSense L53X Recovery'
+    # Take the 'L53' and try to concatenate it 'X' + postfix and than "XX" + postfix
+    # Until find file or exit if not find.
     x = 'X'
+    r = match.regs[1]
```

Comment:
regs?

Reply:
It's undocumented... please use `span(1)`

Reply:
I also think `r` is very unreadable...
Use `start_index, end_index = match.span(1)`

Reply:
done